scrapy/extensions/httpcache.py

```diff
@@ -453,14 +447,34 @@ def _parse_domain_from_url(self, spider, url):
     def _select_source(self, target, sources):
         return sources.keys()[0]
 
-    # Placeholder for now
     def _recompute_deltas(self, new_source, old_source, target_set):
-        #   For each key in the target set:
-        #       - read the delta _data and _length from the db
-        #       - decode the response with the old source
-        #       - encode the response with the new source
-        #       - write the new _data, _time, and _length to db
-        print('Chopper Dave says uh-oh, shouldn\'t hit this yet!\n')
+        for target_key in target_set:
+            # Get old response from the db
+            data = self._read_data(key_to_use=target_key, ignore_time=True)
+            # Decode serialized delta response with old source
+            old_serial_response = self._decode_response(data, old_source)
+            # Deserialze the 
+            data = self._deserialize(old_serial_response)
+            # Reconstruct original response
```

Comment:
encode response cannot perform on an object, it will only accept a read only buffer or a string. This is why i was reconstructing the response object and then serializing it.